src/RestApi/StoreApi/Controllers/CartOrder.php

```diff
@@ -74,141 +100,86 @@ public function register_routes() {
-				// Ignores stock for orders which are no longer drafts (assuming real stock reduction was performed).
-				// Ignores stock reserved over 10 mins ago. Client can call this endpoint to renew holds on stock.
-				$reserved_stock = $wpdb->get_var(
-					$wpdb->prepare(
-						"
-						SELECT SUM( stock_table.`stock_quantity` ) FROM $wpdb->wc_reserved_stock stock_table
-						LEFT JOIN $wpdb->posts posts ON stock_table.`order_id` = posts.ID
-						WHERE stock_table.`product_id` = %d
-						AND posts.post_status = 'wc-draft'
-						AND stock_table.`order_id` != %d
-						AND stock_table.`timestamp` > ( NOW() - INTERVAL 10 MINUTE )
-						",
-						$stocked_product_id,
-						$order->get_id()
-					)
-				);
+	protected function update_session( RestRequest $request ) {
+		$schema = $this->get_item_schema();
 
-				// Deals with legacy stock reservation which the core Woo checkout performs.
-				$reserved_stock += ( $hold_stock_minutes > 0 ) ? wc_get_held_stock_quantity( $product ) : 0;
-
-				if ( ( $product->get_stock_quantity() - $reserved_stock - $stock_to_reserve[ $stocked_product_id ] ) < $item->get_quantity() ) {
-					throw new RestException(
-						'woocommerce_rest_cart_order_product_not_enough_stock',
-						sprintf(
-							/* translators: %s: product name */
-							__( 'Not enough units of %s are available in stock to fulfil this order.', 'woo-gutenberg-products-block' ),
-							$product->get_name()
-						),
-						403
-					);
-				}
-
-				// Queue this reservation for later DB insertion.
-				$stock_to_reserve[ $stocked_product_id ] += $item->get_quantity();
+		if ( isset( $request['billing_address'] ) ) {
+			$allowed_billing_values = array_intersect_key( $request['billing_address'], $schema['properties']['billing_address']['properties'] );
+			foreach ( $allowed_billing_values as $key => $value ) {
+				WC()->customer->{"set_billing_$key"}( $value );
 			}
+		}
 
-			$stock_to_reserve = array_filter( $stock_to_reserve );
-
-			if ( $stock_to_reserve ) {
-				$stock_to_reserve_rows = [];
+		if ( isset( $request['shipping_address'] ) ) {
+			$allowed_shipping_values = array_intersect_key( $request['shipping_address'], $schema['properties']['shipping_address']['properties'] );
+			foreach ( $allowed_shipping_values as $key => $value ) {
+				WC()->customer->{"set_shipping_$key"}( $value );
+			}
+		}
 
-				foreach ( $stock_to_reserve as $product_id => $stock_quantity ) {
-					$stock_to_reserve_rows[] = '(' . esc_sql( $order->get_id() ) . ',"' . esc_sql( $product_id ) . '","' . esc_sql( $stock_quantity ) . '")';
-				}
+		WC()->customer->save();
+	}
 
-				$values = implode( ',', $stock_to_reserve_rows );
+	/**
+	 * Put a temporary hold on stock for this order.
+	 *
+	 * @throws RestException Exception when stock cannot be reserved.
+	 * @param \WC_Order $order Order object.
+	 */
+	protected function reserve_stock( \WC_Order $order ) {
+		$reserve_stock_helper = new ReserveStock();
```

Comment:
Is it okay that this is created every time the method is called?  I wonder if this should be moved to a property on the `CartOrder` and instantiate it on creating that instance?  That way if we eventually refactor things it can be injected via the constructor. As far as I can tell `ReserveStock` doesn't keep state so it would likely be a single instance kept in the DIC.

Reply:
I think the overhead here is pretty low, and it's likely only going to be used/called once per request.

I see how dependency injection patterns can help in some places but I'm not fully on board with injecting things such as utility classes. Also, what happens if a dependency changes? The constructor would then change too right? Maybe a discussion outside of this :)